ir/pattern/conjunction.rs

```diff
@@ -55,6 +55,18 @@ impl Conjunction {
         &mut self.nested_patterns
     }
 
+    pub fn optimise_away(&mut self) {
+        let mut swapped_conjunction = Self::new(self.scope_id);
+        std::mem::swap(&mut self.nested_patterns, &mut swapped_conjunction.nested_patterns);
+        std::mem::swap(
+            self.constraints_mut().constraints_mut(),
+            swapped_conjunction.constraints_mut().constraints_mut(),
+        );
+        self.constraints_mut()
+            .constraints_mut()
+            .push(Constraint::OptimisedToUnsatisfiable(OptimisedToUnsatisfiable::new(swapped_conjunction)))
```

Comment:
You should be able to just assign to `*self`:
```suggestion
        *self = Self::new(self.scope_id);
        self.constraints
            .constraints_mut()
            .push(Constraint::OptimisedToUnsatisfiable(OptimisedToUnsatisfiable::new(swapped_conjunction)))
```

Reply:
I do want the constraints and the nested patterns to be swapped, so I guess you mean `std::mem::swap(self, &mut swapped_conjunction)`?

Reply:
I thought the `swapped_conjunction` was dropped at the end of the function, I missed that it was in `OptimisedToUnsatisfiable::new`.
In that case, you can swap it with `self`, which I can see you've done in a recent commit.